src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -160,6 +160,10 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                     // input value among many constant tests.
                     convertedExpression = operand;
                 }
+                else if (conversion.ConversionKind == ConversionKind.NoConversion && convertedExpression.Type?.IsErrorType() == true)
+                {
+                    convertedExpression = operand;
```

Comment:
Because the converted expression's type is an error type, we've already reported an error. We hit this case when the switch expression (to which the switch case or goto case is converted) is an error, or when the left-hand-side of an is-pattern expression is an error. In those cases we have already reported an error, but it is useful to have the underlying expression of the pattern in the caller so we can, for example, report duplicate case labels, value not constant, etc.

---
In reply to: [76448902](https://github.com/dotnet/roslyn/pull/13382#discussion_r76448902) [](ancestors = 76448902)